torch/optim/zeroptimizer.py

```diff
@@ -0,0 +1,524 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
```

Comment:
Since this is a distributed training feature, let's move this to the distributed package. The folder below is a candidate location.

https://github.com/pytorch/pytorch/tree/83d358da7c9957db6cc6456f69f43a4c2a4d54e4/torch/distributed/optim